AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -54,7 +59,26 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
 }
 
 final class HiddenCoreDataMeasurementStreamStorage: MeasurementStreamStorageContextUpdate {
-
+    func preapareStreamForDeletion(_ sessionUUID: SessionUUID, sensorsName: [String], completion: () -> Void) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        try sensorsName.forEach { sensorName in
+            guard let stream = sessionEntity.allStreams?.first(where: { $0.sensorName == sensorName }) else {
+                Log.info("Problem")
+                return
+            }
+            stream.gotDeleted = true
+            try context.save()
+        }
+        sessionEntity.changesCount += 1
```

Comment:
we should move this to the `try` block right above the `context.save()` (if the "Problem" occurs we don't want to increment)